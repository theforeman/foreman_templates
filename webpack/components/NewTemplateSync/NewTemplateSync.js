import React from 'react';
import { LoadingState } from 'patternfly-react';

import NewTemplateSyncForm from './components/NewTemplateSyncForm';

import './NewTemplateSync.scss';

class NewTemplateSync extends React.Component {
  componentDidMount() {
    const { apiUrls: { syncSettingsUrl }, getSyncSettings } = this.props;
    getSyncSettings(syncSettingsUrl);
  }

  render() {
    const { apiUrls: { importUrl, exportUrl }, loadingSettings, history, validationData } = this.props;

    return (
      <LoadingState loading={loadingSettings}>
        <NewTemplateSyncForm validationData={validationData}
                             importUrl={importUrl}
                             exportUrl={exportUrl}
                             history={history} />
      </LoadingState>
    );
  }
}

export default NewTemplateSync;
