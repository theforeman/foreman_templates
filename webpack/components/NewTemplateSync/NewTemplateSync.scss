@import '~patternfly/dist/sass/patternfly/_variables';
@import '~patternfly/dist/sass/patternfly/_loading-state';
