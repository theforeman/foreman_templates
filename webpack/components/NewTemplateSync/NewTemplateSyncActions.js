import api from 'foremanReact/API';
import { deepPropsToCamelCase } from 'foremanReact/common/helpers';

import {
  SYNC_SETTINGS_REQUEST,
  SYNC_SETTINGS_SUCCESS,
  SYNC_SETTINGS_FAILURE,
  SYNC_RESULT_PAGINATION_CHANGE,
} from '../../consts';

import { initialState } from '../TemplateSyncResult/TemplateSyncResultReducer';

export const getSyncSettings = url => async dispatch => {
  dispatch({ type: SYNC_SETTINGS_REQUEST });

  try {
    const { data } = await api.get(url);
    dispatch({
      type: SYNC_RESULT_PAGINATION_CHANGE,
      payload: { pagination: initialState.pagination },
    });
    return dispatch({
      type: SYNC_SETTINGS_SUCCESS,
      payload: {
        ...deepPropsToCamelCase(data),
      },
    });
  } catch (error) {
    return dispatch(errorHandler(SYNC_SETTINGS_FAILURE, error));
  }
};

const errorHandler = (msg, err) => {
  const error = {
    errorMsg: 'Failed to fetch Settings for template sync from server.',
    statusText: err.response.statusText,
  };
  return { type: msg, payload: { error } };
};
