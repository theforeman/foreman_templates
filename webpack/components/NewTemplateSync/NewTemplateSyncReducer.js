import Immutable from 'seamless-immutable';

import {
  SYNC_SETTINGS_REQUEST,
  SYNC_SETTINGS_SUCCESS,
  SYNC_SETTINGS_FAILURE,
} from '../../consts';

export const initialState = Immutable({
  loadingSettings: false,
  importSettings: [],
  exportSettings: [],
  error: '',
});

const syncSettings = (state = initialState, action) => {
  const { payload } = action;
  switch (action.type) {
    case SYNC_SETTINGS_REQUEST:
      return state.set('loadingSettings', true);
    case SYNC_SETTINGS_SUCCESS:
      return state.merge({
        loadingSettings: false,
        importSettings: payload.results.import,
        exportSettings: payload.results.export,
      });
    case SYNC_SETTINGS_FAILURE:
      return state.merge({ error: payload.error, loadingSettings: false });
    default:
      return state;
  }
};

export default syncSettings;
