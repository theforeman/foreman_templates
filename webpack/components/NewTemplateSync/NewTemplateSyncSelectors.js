export const newSyncState = state => state.foremanTemplates.syncSettings;

export const selectImportSettings = state => newSyncState(state).importSettings;
export const selectExportSettings = state => newSyncState(state).exportSettings;
export const selectLoadingSettings = state =>
  newSyncState(state).loadingSettings;
export const selectError = state => newSyncState(state).error;
