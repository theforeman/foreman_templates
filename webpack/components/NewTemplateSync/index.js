import React from 'react';
import { connect } from 'react-redux';

import * as TemplateSyncActions from './NewTemplateSyncActions';
import NewTemplateSync from './NewTemplateSync';
import { selectLoadingSettings, selectError } from './NewTemplateSyncSelectors';
import withProtectedView from '../../withProtectedView';
import PermissionDenied from '../PermissionDenied';

const mapStateToProps = state => ({
  loadingSettings: selectLoadingSettings(state),
  error: selectError(state),
});

const permissionList = (
  <ul className="list-unstyled">
    <li>import_templates</li>
    <li>export_templates</li>
  </ul>
);

export default withProtectedView(
  connect(mapStateToProps, TemplateSyncActions)(NewTemplateSync),
  PermissionDenied,
  props =>
    props.userPermissions &&
    (props.userPermissions.import || props.userPermissions.export),
  { doc: permissionList }
);
