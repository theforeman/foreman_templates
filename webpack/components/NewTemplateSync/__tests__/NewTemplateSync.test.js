import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import NewTemplateSync from '../NewTemplateSync';

jest.mock('foremanReact/pages/common/PageLayout/PageLayout');

const noop = () => {};

const commonFixture = {
  apiUrls: {},
  getSyncSettings: noop,
  validationData: {},
  userPermissions: {
    import: true,
    export: true,
  },
};

const fixtures = {
  'should render when loaded': {
    loadingSettings: false,
    ...commonFixture,
  },
  'should render when loading': {
    loadingSettings: true,
    ...commonFixture,
  },
};

describe('NewTemplateSync', () =>
  testComponentSnapshotsWithFixtures(NewTemplateSync, fixtures));
