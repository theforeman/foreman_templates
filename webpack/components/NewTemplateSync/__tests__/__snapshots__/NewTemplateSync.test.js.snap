// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTemplateSync should render when loaded 1`] = `
<LoadingState
  additionalClasses=""
  loading={false}
  loadingText="Loading"
  size="lg"
  timeout={300}
>
  <PageLayout
    header="Import or Export Templates"
    searchable={false}
  >
    <Connect(NewTemplateSyncForm)
      history={Object {}}
      userPermissions={
        Object {
          "export": true,
          "import": true,
        }
      }
      validationData={Object {}}
    />
  </PageLayout>
</LoadingState>
`;

exports[`NewTemplateSync should render when loading 1`] = `
<LoadingState
  additionalClasses=""
  loading={true}
  loadingText="Loading"
  size="lg"
  timeout={300}
>
  <PageLayout
    header="Import or Export Templates"
    searchable={false}
  >
    <Connect(NewTemplateSyncForm)
      history={Object {}}
      userPermissions={
        Object {
          "export": true,
          "import": true,
        }
      }
      validationData={Object {}}
    />
  </PageLayout>
</LoadingState>
`;
